packages/curl.rb

```diff
@@ -3,108 +3,237 @@
-  depends_on 'zlibpkg' # R
-  depends_on 'zstd' # R
 
   def self.build
     raise StandardError, 'Please remove libiconv before building.' if File.exist?("#{CREW_LIB_PREFIX}/libcharset.so")
 
-    @krb5_static_libs = "#{CREW_LIB_PREFIX}/libkrb5support.a " + "#{CREW_LIB_PREFIX}/libgssapi_krb5.a " + "#{CREW_LIB_PREFIX}/libkrb5.a " + "#{CREW_LIB_PREFIX}/libk5crypto.a " + "#{CREW_LIB_PREFIX}/libcom_err.a "
+    @krb5_static_libs = '-l:libkrb5support.a -l:libgssapi_krb5.a -l:libkrb5.a -l:libk5crypto.a -l:libcom_err.a'
 
-    FileUtils.mkdir 'ssl'
-    FileUtils.ln_s "#{CREW_PREFIX}/include/openssl", 'ssl/include'
-    FileUtils.ln_s CREW_LIB_PREFIX, 'ssl/lib'
-    @sslpath = "#{`pwd`.chomp}/ssl"
+    FileUtils.mkdir_p 'build'
+    FileUtils.mkdir_p 'deproot/include'
+    FileUtils.mkdir_p 'deproot/lib'
+    Dir.chdir 'deproot' do
+      FileUtils.ln_s 'lib', 'lib64'
+    end
+    @deppath = "#{`pwd`.chomp}/deproot"
+    FileUtils.ln_s "#{CREW_PREFIX}/include/linux", "#{@deppath}/include/"
+    FileUtils.ln_s "#{CREW_PREFIX}/include/asm", "#{@deppath}/include/"
+    FileUtils.ln_s "#{CREW_PREFIX}/include/asm-generic", "#{@deppath}/include/"
+
+    @arch_ssp_cflags = ''
+    case ARCH
+    when 'i686'
+      @arch_ssp_cflags = '-fno-stack-protector'
+    end
+
+    @curldep_cmake_options = "CC=musl-gcc \
+          CFLAGS='-flto -pipe -O3 -fPIC -ffat-lto-objects -fipa-pta -fno-semantic-interposition -fdevirtualize-at-ltrans #{@arch_ssp_cflags}' \
+          CXXFLAGS='-flto -pipe -O3 -fPIC -ffat-lto-objects -fipa-pta -fno-semantic-interposition -fdevirtualize-at-ltrans #{@arch_ssp_cflags}' \
+          CPPFLAGS='-I#{@deppath}/include' \
+          LDFLAGS='-L#{@deppath}/#{ARCH_LIB}' \
+          cmake \
+          -DCMAKE_INSTALL_PREFIX='#{@deppath}' \
+          -DCMAKE_LIBRARY_PATH='#{@deppath}/#{ARCH_LIB}' \
+          -DCMAKE_C_COMPILER=`which musl-gcc` \
+          -DCMAKE_INCLUDE_DIRECTORIES_BEFORE=ON \
+          -DINCLUDE_DIRECTORIES=#{@deppath}/include \
+          -DCMAKE_C_FLAGS='-O3 -fPIC -pipe -flto -ffat-lto-objects -fipa-pta -fno-semantic-interposition -fdevirtualize-at-ltrans #{@arch_ssp_cflags}' \
+          -DCMAKE_EXE_LINKER_FLAGS='-flto' \
+          -DCMAKE_SHARED_LINKER_FLAGS='-flto' \
+          -DCMAKE_STATIC_LINKER_FLAGS='-flto' \
+          -DCMAKE_MODULE_LINKER_FLAGS='-flto' \
+          -DPROPERTY_INTERPROCEDURAL_OPTIMIZATION=TRUE \
+          -DCMAKE_BUILD_TYPE=Release"
+
+    @curldep_env_options = "CC=musl-gcc \
+        CFLAGS='-flto -pipe -O3 -fPIC -ffat-lto-objects -fipa-pta -fno-semantic-interposition -fdevirtualize-at-ltrans #{@arch_ssp_cflags}' \
+        CXXFLAGS='-flto -pipe -O3 -fPIC -ffat-lto-objects -fipa-pta -fno-semantic-interposition -fdevirtualize-at-ltrans#{@arch_ssp_cflags}' \
+        CPPFLAGS='-I#{@deppath}/include' \
+        LDFLAGS='-L#{@deppath}/#{ARCH_LIB}'"
+
+    FileUtils.mkdir_p 'build/zlib'
+    Dir.chdir 'build/zlib' do
+      puts 'Building Zlib.'.lightgreen
```

Comment:
The color green typically indicates success.  Would yellow be better for a status message?

Reply:
yes